pkg/endpoint/bpf.go

```diff
@@ -337,31 +338,62 @@ func (e *Endpoint) writeHeaderfile(prefix string, owner Owner) error {
 	return fw.Flush()
 }
 
-// hashHeaderfile returns the MD5 hash of the BPF headerfile with the given prefix.
+// hashEndpointHeaderFiles returns the MD5 hash of any header files that are
+// used in the compilation of an endpoint's BPF program. Currently, this
+// includes the endpoint's headerfile, and the node's headerfile.
+func hashEndpointHeaderfiles(prefix string) (string, error) {
+	endpointHeaderPath := filepath.Join(prefix, common.CHeaderFileName)
+	nodeConfigPath := "./globals/node_config.h"
```

Comment:
I just realized that this has to be changed to be an absolute path - you will need to pass `Owner` to this function and then call `owner.GetStateDir()` to get `nodeConfigPath` to be at the correct path. Sorry about that :(

Reply:
Actually, if https://github.com/cilium/cilium/pull/4410/commits/f7e4adf68ec0718962a8f4c3fb0782fe2c739ff3 was a backport as well to 1.0 (i.e., we are not hashing the headerfiles in 1.0), then we do not need to backport this commit either to 1.0, only 1.1. I believe you can revert these two commits. The hashing of the headerfile was an optimization, and this commit  ( https://github.com/cilium/cilium/pull/4410/commits/58a38a5523df35a8aa09a72a6de2350b07cffa0d ) is a fix for that commit which you also had to backport. cc: @rlenglet to confirm I am correct. Sorry about all of the confusion.